pkg/nodeconfig/nodeconfig.go

```diff
@@ -258,8 +258,8 @@ func (nc *nodeConfig) SafeReboot(ctx context.Context, node *core.Node) error {
 	if err := nc.Windows.RebootAndReinitialize(); err != nil {
 		return err
 	}
-	// wait for WICD to remove Reboot annotation
-	if err := metadata.WaitForRebootAnnotationRemoval(ctx, nc.client, nc.node.Name); err != nil {
+	// Remove the reboot annotation after we know the reboot occurred successfully
+	if err := metadata.RemoveRebootAnnotation(ctx, nc.client, *nc.node); err != nil {
```

Comment:
Do you really need to dereference the pointer here as per https://go.dev/ref/spec#Selectors?

Reply:
The deference is needed unless we change RemoveVersionAnnotation()'s signature to expect pointer